jumanji/environments/routing/sokoban/env.py

```diff
@@ -0,0 +1,593 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional, Sequence, Tuple
+
+import chex
+import jax
+import jax.numpy as jnp
+import matplotlib.animation
+
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.environments.routing.sokoban.constants import (
+    AGENT,
+    BOX,
+    EMPTY,
+    GRID_SIZE,
+    MOVES,
+    N_BOXES,
+    NOOP,
+    TARGET,
+    TARGET_AGENT,
+    TARGET_BOX,
+    WALL,
+)
+from jumanji.environments.routing.sokoban.generator import (
+    Generator,
+    HuggingFaceDeepMindGenerator,
+)
+from jumanji.environments.routing.sokoban.reward import DenseReward, RewardFn
+from jumanji.environments.routing.sokoban.types import Observation, State
+from jumanji.environments.routing.sokoban.viewer import BoxViewer
+from jumanji.types import TimeStep, restart, termination, transition
+from jumanji.viewer import Viewer
+
+
+class Sokoban(Environment[State]):
+    """A JAX implementation of the 'Sokoban' game from deepmind.
+
+    - observation: `Observation`
+        - grid: jax array (uint8) of shape (num_rows, num_cols, 2)
+            Array that includes information about the agent, boxes, and
+            targets in the game.
+        - step_count: jax array (int32) of shape ()
+            current number of steps in the episode.
+
+    - action: jax array (int32) of shape ()
+        [0,1,2,3] -> [Up, Right, Down, Left].
+
+    - reward: jax array (float) of shape ()
+        A reward of 1.0 is given for each box placed on a target and -1
+        when removed from a target and -0.1 for each timestep.
+        10 is awarded when all boxes are on targets.
+
+    - episode termination:
+        - if the time limit is reached.
+        - if all boxes are on targets.
+
+    - state: `State`
+        - key: jax array (uint32) of shape (2,) used for auto-reset
+        - fixed_grid: jax array (uint8) of shape (num_rows, num_cols)
+            array indicating the walls and targets in the level.
+        - variable_grid: jax array (uint8) of shape (num_rows, num_cols)
+            array indicating the current location of the agent and boxes.
+        - agent_location: jax array (int32) of shape (2,)
+            the agent's current location.
+        - step_count: jax array (int32) of shape ()
+            current number of steps in the episode.
+
+    ```python
+    from jumanji.environments import Sokoban
+    from jumanji.environments.routing.sokoban.generator import
+    HuggingFaceDeepMindGenerator,
+
+    env_train = Sokoban(
+        generator=HuggingFaceDeepMindGenerator(
+            dataset_name="unfiltered-train",
+            proportion_of_files=1,
+        )
+    )
+
+    env_test = Sokoban(
+        generator=HuggingFaceDeepMindGenerator(
+            dataset_name="unfiltered-test",
+            proportion_of_files=1,
+        )
+    )
+
+    #Train...
+
```

Comment:
Nit, you were missing the three tilde's above this block: 
```suggestion
    # Train...
    
    ```
```